### PR TITLE
Add whitepaper v1.0 in markdown

### PR DESCRIPTION
**Pull request by caniszczyk**
Signed-off-by: Chris Aniszczyk <caniszczyk@gmail.com>

### CODE REVIEW

**Comment by caniszczyk** [comment]
RFC @duglin for sanity check, it would be great to get this launched for Wed

**Comment by caniszczyk** [comment]
NOTE: if I don't hear any objections, I'll merge this in EOD tomorrow before the press release on Wed

File: whitepaper/README.md [hunk 1]
```diff
@@ -0,0 +1,1008 @@
...
+
+## Serverless vs. Other Cloud Native Technologies
+
+There are three primary development and deployment models that most application developers might consider when looking for a platform to host their cloud-native applications. Each model has its own set of differing implementations (whether an open source project, hosted platform, or on-premises product). These three models are commonly built upon container technology for its density, performance, isolation, and packaging features, but containerization is not a requirement.
+
+In order of increasing abstraction away from the actual infrastructure that is running their code, and toward a greater focus on only the business logic that’s developed, they are Container Orchestration (or Containers-as-a-Service), Platform-as-a-Service, and Serverless (Functions-as-a-Service).* *All of these approaches provide a way to deploy cloud-native application, but they prioritize different functional and non-functional aspects based on their intended developer audience and workload type. The following section lists some of the key characteristics of each.
+
```
Comment (justincormack) [inline whitepaper/README.md]: formatting issues around "All of these approaches provide a way to deploy "

File: whitepaper/README.md [hunk 2]
```diff
@@ -0,0 +1,1008 @@
...
+![function operations](image_2.png)
+
+When creating a function, providing its metadata (as described later under function spec) as part of the function creation, it will be compiled and possibly published. A function may be started, disabled and enabled later on. Function deployments need to be able to support the following usecases: 
+
+* Event streaming, in this use case there may always be events in queue however the processing may need to be paused/resumed through an explicit request
+
+* Warm startup - function that has minimal number of instances at any time, such that the "first" event received has a warm start since the function is already deployed and is ready to serve the event (as opposed to a cold start where the function gets deployed on the first invocation by an “incoming” event) 
+
+A user can **Publish **a function, this will create a new version (copy of the "latest" version), the published version may be tagged/labeled or have aliases, see more below.
```
Comment (justincormack) [inline whitepaper/README.md]: spacing around `**` breaks formatting

File: whitepaper/README.md [hunk 3]
```diff
@@ -0,0 +1,1008 @@
...
+
+* Event streaming, in this use case there may always be events in queue however the processing may need to be paused/resumed through an explicit request
+
+* Warm startup - function that has minimal number of instances at any time, such that the "first" event received has a warm start since the function is already deployed and is ready to serve the event (as opposed to a cold start where the function gets deployed on the first invocation by an “incoming” event) 
+
+A user can **Publish **a function, this will create a new version (copy of the "latest" version), the published version may be tagged/labeled or have aliases, see more below. 
+
+A user may want to **Execute/Invoke** a function directly (bypass the event source or API gateway) for debug and development processes. A user may specify invocation parameters such as desired version, Sync/Async operation, Verbosity level, etc. 
+
+Users may want to obtain function **Statistics **(e.g. number of invocations, average runtime, average delay, failures, retries, etc.), statistics can be the current metric values or a time-series of values (e.g. stored in Prometheus or cloud provider facility such as AWS Cloud Watch).
```
Comment (justincormack) [inline whitepaper/README.md]: also `**` issues

File: whitepaper/README.md [hunk 4]
```diff
@@ -0,0 +1,1008 @@
...
+
+### Function Versioning and Aliases
+
+A Function may have multiple versions, providing the user the ability to run different level of codes such as beta/production, A/B testing etc. When using versioning, the function version is "latest" by default, the “latest” version can be updated and modified, potentially triggering a new build process on every such change. 
+
+Once a user wants to freeze a version he will use a **Publish **operation that will create a new version with potential tags or aliases (e.g. "beta", “production”) to be used when configuring an event source, so an event or API call can be routed to a specific function version. Non-latest function versions are immutable (their code and all or some of the function spec) and cannot be changed once published; functions cannot be “un-published” instead they should be deleted.
```
Comment (justincormack) [inline whitepaper/README.md]: ditto here

File: whitepaper/README.md [hunk 5]
```diff
@@ -0,0 +1,1008 @@
...
+
+    * Usually a stream is sharded to multiple partitions/shards with a single worker (the shard consumer) per shard 
+
+    * Stream can be produced from messages, database updates (journal), or files (e.g. CSV, Json, Parquet) 
+
+    * Events can be pushed into the function runtime or pulled by the function runtime
+
+4. **Batch Jobs** e.g. ETL jobs, distributed deep learning, HPC simulation 
+
+    * Jobs are scheduled or submitted to a queue, and processed at run time using **multiple **function instances in **parallel**, each handling one or more portion of the working set (a task)
```
Comment (justincormack) [inline whitepaper/README.md]: spacing issue with `**` here breaks emphasis

File: whitepaper/README.md [hunk 6]
```diff
@@ -0,0 +1,1008 @@
...
+
+* Network Policy / VPC 
+
+* Data Bindings 
+
+### Metadata details 
+
+Function frameworks may include the following metadata for functions:
+
+* **Version**** **- each function version should have a unique identifier, in addition versions can be labeled using one or more aliases (e.g. "latest", “production”, “beta”). API gateways and event sources would route traffic/events to a specific function version.
```
Comment (justincormack) [inline whitepaper/README.md]: emphasis issues...

File: whitepaper/README.md [hunk 7]
```diff
@@ -0,0 +1,1008 @@
...
+* **Execution Role** - the function should run under a specific user or role identity that grants and audits its access to platform resources.
+
+* **Resources **- define the required or maximum hardware resources such as Memory and CPU used by the function.
+
+* **Timeout **- specify the maximum time a function call can run until it is terminated by the platform.
+
+* **Failure Log (Dead Letter Queue)** - a path to a queue or stream that will store the list of failed function executions with appropriate details.
+
+* **Network Policy **- the network domain and policy assigned to the function (for the function to communicate with external services/resources).
+
```
Comment (justincormack) [inline whitepaper/README.md]: also above here...

File: whitepaper/README.md [hunk 8]
```diff
@@ -0,0 +1,1008 @@
...
+However, serverless technology is not a perfect fit for all cases and careful consideration should be given to when it is appropriate.Short-lived, event-driven processing is driving early adoption and use cases for businesses that expect a high rate of change with unpredictable capacity and infrastructure needs are emerging. See the [Additional References](#heading=h.nkn4basctyj) section for more reading material and insights into serverless computing.
+
+The CNCF Servleress Working Group, in partnership with [Redpoint Ventures](http://www.redpoint.com/), recently published a a [Serverless Landscape](https://docs.google.com/spreadsheets/d/10rSQ8rMhYDgf_ib3n6kfzwEuoE88qr0amUPRxKbwVCk/edit#gid=0). It illustrates some of the major erverless projects, tooling and services that are available in the ecosystem. It is not intended to represent a comprehensive, fully inclusive serverless ecosystem, nor is it an endorsement, rather just an overview of the landscape.. It is expected that owners of each will provide updates in an attempt to keep it up to date.
+
```
Comment (justincormack) [inline whitepaper/README.md]: "erverless" above

**Pull request merged**